compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -153,10 +153,40 @@ public SymbolEnv evaluateTruth(BLangExpression expr, BType typeToRemove, BLangNo
         SymbolEnv targetEnv = getTargetEnv(targetNode, env);
         BVarSymbol originalVarSym = getOriginalVarSymbol((BVarSymbol) varRef.symbol);
         symbolEnter.defineTypeNarrowedSymbol(varRef.pos, targetEnv, originalVarSym, remainingType,
-                originalVarSym.origin == VIRTUAL);
+                originalVarSym.origin == VIRTUAL, false);
         return targetEnv;
     }
 
+    public SymbolEnv evaluateFalsityForSingleIf(BLangExpression expr, SymbolEnv currentEnv) {
```

Comment:
From the method name (or probably from the lack of comments) it's not clear how it's different from normal evaluateFalsity?